src/app/public/modules/file-attachment/file-item.component.spec.ts

```diff
@@ -0,0 +1,232 @@
+    let imageEl = getImage();
+    expect(imageEl).toBeFalsy();
+  }
+  //#endregion
+
+  it('shows the name and size if the item is a file', () => {
+
+    componentInstance.fileItem = <SkyFileItem>{
+      file: {
+        name: 'myFile.txt',
+        size: 1000
+      }
+    };
+    fixture.detectChanges();
+
+    let nameEl = getNameEl();
+
+    expect(nameEl.nativeElement.textContent).toBe('myFile.txt');
+
+    let sizeEl = getSizeEl();
+    expect(sizeEl.nativeElement.textContent).toContain('(1 KB)');
+  });
+
+  it('shows the url if the item is a link', () => {
+
+    componentInstance.fileItem = <SkyFileLink>{
+      url: 'myFile.txt'
+    };
+
+    fixture.detectChanges();
+
+    let nameEl = getNameEl();
+
+    expect(nameEl.nativeElement.textContent).toBe('myFile.txt');
+
+    let sizeEl = getSizeEl();
+    expect(sizeEl).toBeFalsy();
+  });
+
+  it('emits the delete event when the delete button is clicked', () => {
+    componentInstance.fileItem = <SkyFileLink>{
+      url: 'myFile.txt'
+    };
+    let deletedItem: SkyFileLink;
+
+    componentInstance.deleteFile.subscribe(
+      (newDeletedFile: SkyFileLink) => deletedItem = newDeletedFile );
+
+    fixture.detectChanges();
+    triggerDelete();
+
+    expect(deletedItem.url).toBe('myFile.txt');
+
+    componentInstance.fileItem = <SkyFileItem>{
+      file: {
+        name: 'myFile.txt',
+        size: 1000,
+        type: 'file/txt'
+      }
+    };
+
+    let deletedFile: SkyFileItem;
+
+    componentInstance.deleteFile.subscribe(
+      (newDeletedFile: SkyFileItem) => deletedFile = newDeletedFile );
+    fixture.detectChanges();
+
+    triggerDelete();
+
+    expect(deletedFile.file.name).toBe('myFile.txt');
+    expect(deletedFile.file.size).toBe(1000);
+  });
+
+  it('shows an image if the item is an image', () => {
+    testImage('png');
+    testImage('bmp');
+    testImage('jpeg');
+    testImage('gif');
+  });
+
+  it('shows a file icon with the proper extension if it is not an image', () => {
+    testOtherPreview('pdf', 'pdf');
+    testOtherPreview('gz', 'gz');
+    testOtherPreview('rar', 'rar');
+    testOtherPreview('tgz', 'tgz');
+    testOtherPreview('zip', 'zip');
+    testOtherPreview('ppt', 'ppt');
+    testOtherPreview('pptx', 'pptx');
+    testOtherPreview('doc', 'doc');
+    testOtherPreview('docx', 'docx');
+    testOtherPreview('xls', 'xls');
+    testOtherPreview('xlsx', 'xlsx');
+    testOtherPreview('txt', 'txt');
+    testOtherPreview('htm', 'htm');
+    testOtherPreview('html', 'html');
+    testOtherPreview('mp3', 'audio');
+    testOtherPreview('tiff', 'image');
+    testOtherPreview('other', 'text');
+    testOtherPreview('mp4', 'video');
+  });
+
+  it('should pass accessibility', async(() => {
+    componentInstance.fileItem = <SkyFileItem>{
+      file: {
+        name: 'myFile.txt',
+        size: 1000
+      }
+    };
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      expect(fixture.nativeElement).toBeAccessible();
```

Comment:
If possible, could you sprinkle `toBeAccessible` expectations throughout the other tests, to make sure the various states are passing accessibility as well?

Reply:
I added a couple accessibility checks to the helper functions that run each time a file or image is checked. There's also one in there for the URL test. Also, good thing you brought this up - it caught an existing missing alt tag!